TACC/tup-ui
# TUP 380 Create blank proj & allocations page

## Overview:

Create a blank projects page as a workspace for projects/allocations component development.

## Related:

- [TUP-380](https://jira.tacc.utexas.edu/browse/TUP-380)

## Changes:

- Added route in the tup-ui app for projects
- Created project page and index for page in tup-ui app/projects
- Created projectslayout to handle the rendering of page component in tup-components with a unit test
- Originally had the dashboard table rendering on the PR but removed it. Some changes to projectTable from working with that to streamline the UI

## Testing:

1. Go to [localhost:8001/dashboard](http://localhost:8000/dashboard)
2. Click on Projects & Allocations in the side bar or the button above the projects table
3. Make sure link goes to a page with a title "Projects & Allocations"
4. Go to [localhost:8000/dashboard/projects](http://localhost:8000/dashboard/projects) and make sure a page with title "Projects & Allocations" appears

## UI:
Changed the top div in the ProjectsTable to match the specifications according to Wesley's UI doc [here](https://dev.tup.tacc.utexas.edu/static/ui/index.html). Let me know if I'm missing or something or missed the mark. 

I made it so the top div is a "o-fixed-header-table" and removed infinitescrolltable class. 

@wesleyboar If you have time to do a quick check of the ProjectsLayout component, passed to tup-ui pages then to the main App to make sure I'm using your UI correctly, I'd appreciate it!


<img width="1776" alt="Screen Shot 2022-12-05 at 4 13 12 PM" src="https://user-images.githubusercontent.com/96220951/205753738-cbb9a8e6-c5e3-4ebf-90c6-91e2855afd57.png">

## Notes:
